package.json

```diff
@@ -70,7 +70,7 @@
     "start": "react-scripts start",
     "prestart:customise": "cpx 'src/**/*' tmp",
     "start:customise": "cpx \"$CUSTOMISATIONS_DIR/**/*\" src/public --watch & yarn start",
-    "build": "REACT_APP_BUILD_ID=\"$(nanoid)\" GENERATE_SOURCEMAP=false REACT_APP_REVISION=\"$(git rev-parse --short HEAD)\" react-scripts build",
+    "build": "REACT_APP_BUILD_ID=\"$(nanoid)\" GENERATE_SOURCEMAP=false REACT_APP_REVISION=\"${REACT_APP_REVISION:-$(git rev-parse --short HEAD)}\" react-scripts build",
```

Comment:
@fedde-s what is the reasoning for prepending the commit hash with "REACT_APP_REVISION:-"?

Reply:
That's not what this is supposed to do—`${varname:-word}` is [POSIX-standard shell syntax](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02) for using the value of the variable if it's set and non-blank, or defaulting to `word` otherwise. Effectively, this uses `$REACT_APP_REVISION` if it's supplied (by the environment defined in the Dockerfile), but expands `$(git rev-parse --short HEAD)` as usual if it isn't.

Reply:
I must confess I haven't actually looked up documentation on what interpreter yarn is using to run this though—I assumed it would be a roughly standard shell like sh or bash because of the way the command already used `=` and `$()`.

Reply:
Also, I built images from the Dockerfile on multiple GNU/Linux systems, and the syntax was parsed as I expected there.

Reply:
The [relevant yarn documentation](https://classic.yarnpkg.com/en/docs/cli/run/) is completely implicit on how it parses the script, and [the corresponding npm subcommand](https://docs.npmjs.com/cli/run-script#description) does it in a platform-dependent way, specifically `cmd.exe` on Windows. I am not familiar enough with `cmd.exe` to know whether the syntax I introduce breaks something that worked with the original `varname=$(command) command`.

Reply:
Thanks @fedde-s for the thorough explanation, I've learned something new 👍